src/components/themed/FilledTextInput.tsx

```diff
@@ -237,6 +238,7 @@ export const FilledTextInput = React.forwardRef<FilledTextInputRef, FilledTextIn
     if (onBlur != null) onBlur()
   })
   const handleChangeText = useHandler((value: string) => {
+    valueRef.current = value
```

Comment:
In the opposite direction, do we want an effect like this:

```ts
React.useEffect(() => {
  if (value !== valueRef.current) {
    setNativeProps({ value })
    valueRef.current = value
  }
}, [value])
```

That way, if somebody does `<FilledTextInput value={somethingUnexpected} />`, the text content will update. For normal typing, where the parent just catches the `onChangeText` value and sends it back into the `value` prop, this effect will be a no-op, since `valueRef` will already anticipate the new `value`, and be equal to it. On the other hand, if the parent manually sets the value to something different, such as `onMaxPress = () => setValue(maxBalance)`, the text input will properly update.

Edit: Oh, you already do this in the other component. Please copy it here too.